var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -140,38 +158,67 @@ class Babelstream(CMakePackage, CudaPackage, ROCmPackage):
     variant("flags", values=str, default="none", description="Additional CXX flags to be provided")
 
     # comp_impl_vals=["ONEAPI-DPCPP","DPCPP","HIPSYCL","COMPUTECPP"]
-    variant(
-        "implementation",
-        values=str,
-        default="none",
-        description="Compile using the specified SYCL compiler option",
-    )
-
-    conflicts(
-        "implementation=none",
-        when="+sycl",
-        msg="SYCL requires compiler implementation to be specified by option=",
-    )
-    conflicts(
-        "implementation=none",
-        when="+thrust",
-        msg="Which Thrust implementation to use, supported options include:\
-         - CUDA (via https://github.com/NVIDIA/thrust)\
-         - ROCM (via https://github.com/ROCmSoftwarePlatform/rocThrust)",
-    )
+    with when("build_system=cmake"):
+        variant(
+            "implementation",
+            values=str,
+            default="none",
+            description="Compile using the specified SYCL compiler option",
+        )
+        conflicts(
+            "implementation=none",
+            when="+sycl",
+            msg="SYCL requires compiler implementation to be specified by option=",
+        )
+        conflicts(
+            "implementation=none",
+            when="+thrust",
+            msg="Which Thrust implementation to use, supported options include:\
+            - CUDA (via https://github.com/NVIDIA/thrust)\
+            - ROCM (via https://github.com/ROCmSoftwarePlatform/rocThrust)",
+        )
+        depends_on("cmake@3.14.0:", type="build")
 
     # This applies to all
-    depends_on("cmake@3.14.0:", type="build")
-    depends_on("opencl-c-headers", when="+ocl")
 
+    depends_on("opencl-c-headers", when="+ocl")
+    implementation_vals_cmake = ["cuda", "rocm", "oneapi-dpcpp", "dpcpp", "hipsycl", "computecpp"]
+
+    with when("build_system=makefile"):
+        implementation_vals = [
+            "DoConcurrent",
+            "Array",
+            "OpenMP",
+            "OpenMPWorkshare",
+            "OpenMPTarget",
+            "OpenMPTargetLoop",
+            "OpenMPTaskloop",
+            "OpenACC",
+            "OpenACCArray",
+            "CUDA",
+            "CUDAKernel",
+            "Sequential",
+        ]
+        variant(
+            "impl", values=implementation_vals, default="Sequential", description="Implementation"
```

Comment:
Is this fundamentally the same `implementation` variant as above, or something else? It's not clear what this is doing from the code